plugins/cpp/service/src/cppservice.cpp

```diff
@@ -1238,7 +1238,7 @@ void CppServiceHandler::getFileDiagramTypes(
     return_["This module depends on"]               = EXTERNAL_DEPENDENCY;
     return_["Users of this module"]                 = EXTERNAL_USERS;
   }
-  else
+  else if (file->parseStatus != model::File::PSNone)
```

Comment:
```suggestion
  else if (file->type == "CPP")
```
This way we could also filter those files which are parsed by another plugin.